src/main/java/pl/datamatica/traccar/model/RoutePoint.java

```diff
@@ -46,9 +46,20 @@
 
     public RoutePoint() {}
     
+    public RoutePoint(long id, GeoFence geofence) {
+        this.id = id;
+        this.geofence = geofence;
+    }
+    
     public long getId() {
         return id;
     }
+    
+    public void setId(long id) {
+        if(this.id != 0)
+            throw new IllegalStateException();
```

Comment:
j.w.
Czy to dobry rodzaj wyjątku?

Reply:
Myślę, że tak. Obiekt jest w niepoprawnym stanie, żeby wywołać tę metodę.